src/routes/api.ts

```diff
@@ -0,0 +1,53 @@
+import express, { Router } from 'express';
+import passport from 'passport';
+import { Strategy as BearerStrategy } from 'passport-http-bearer';
+import { RifmDAO } from '../data/rifmDAO';
+import { SpotifyDAO } from '../data/spotifyDAO';
+import { User, UserDAO } from '../data/userDAO';
+
+export function getRouter(spotify: SpotifyDAO, rifm: RifmDAO, users: UserDAO): Router {
+    const router = express.Router();
+
+    passport.use(new BearerStrategy(
+        (token: string, done: Function) => {
+            users.getUserFromToken(token)
+            .then((user: User) => {
+                if (!user) { return done(null, false); }
+                return done(null, user);
+            })
+            .catch((err: any) => {
+                return done(err);
+            });
+        }
+    ));
+
+    router.get('/spotify/search',
+        passport.authenticate('bearer', { session: false }),
+        (req, res) => {
+            const searchTerm = req.query.q;
+            spotify.searchAlbums(searchTerm)
+            .then((results) => res.json(results))
+            .catch((err) => res.status(500).send(err));
+        }
+    );
+    router.get('/rifm/albums/:month',
+        passport.authenticate('bearer', { session: false }),
+        (req, res) => {
+            const month = req.params.month;
+            rifm.getAlbumsForMonth(month)
+            .then((results) => res.json(results))
+            .catch((err) => res.status(500).send(err));
```

Comment:
Might wanna double check that the errors being thrown by the RIFM class aren't going to leak info in the response (details about dynamo, stack traces with config values, etc.). I'd typically log the real error internally and then return a guaranteed safe error to the client

Reply:
Yeah error handling is in mind backlog hahaha